src/Servers/IIS/IIS/src/Core/IISHttpContext.IO.cs

```diff
@@ -198,14 +272,27 @@ internal void AbortIO(bool clientDisconnect)
 
             _bodyOutput.Dispose();
 
-            var cts = _abortedCts;
-            if (cts != null)
+            if (shouldScheduleCancellation)
             {
+                // Potentially calling user code. CancelRequestAbortedToken logs any exceptions.
                 ThreadPool.QueueUserWorkItem(t =>
```

Comment:
Nit: I know this isn't new, but why doesn't this use UnsafeQueueUserWorkItem and pass this as the state variable?

Reply:
Probably should 😄